### DISCUSSION TITLE
How to solve clean code with sonar lint

### DISCUSSION THREAD

**Discussion by dhamdani666** [spatie/laravel-data#649] [Q&A]
I use laravel-data for my project to make it easier for me to transfer object data, but there is a problem when I save some parameters, here is my code
<pre><code> public function __construct(
        public string|null $partnerServiceId,
        public string|null $customerNo,
        public string|null $virtualAccountNo,
        public string|null $trxDateInit,
        public string|null $channelCode,
        public string|null $sourceBankCode,
        public string|null $inquiryRequestId,
        public array|null  $additionalInfo
    )
    {
    }
</code></pre>

This function has 8 parameters, which is greater than the 7 authorized. , what is the solution, do I have to create a new dto then do child , or is there a good solution?

**Comment by rubenvanassche** [comment]
Add as many paramaters as you want, I would not say this is a good idea for other methods. But in the case of a data constructor, you're defining properties. There's no limit on how many properties you should be able to define.
